src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -0,0 +1,226 @@
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.VerificationException;
+import co.rsk.bitcoinj.crypto.TransactionSignature;
+import co.rsk.bitcoinj.params.MainNetParams;
+import co.rsk.bitcoinj.script.RedeemScriptParser.MultiSigType;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+
+public class FlyoverRedeemScriptParserTest {
+
+    private final List<BtcECKey> defaultRedeemScriptKeys = RedeemScriptUtils.getDefaultRedeemScriptKeys();
+    private final List<BtcECKey> emergencyRedeemScriptKeys = RedeemScriptUtils.getEmergencyRedeemScriptKeys();
+    private final Sha256Hash derivationArgumentsHash = Sha256Hash.of(new byte[]{1});
+    private Script standardRedeemScript;
+    private Script flyoverStandardRedeemScript;
+    private Script erpRedeemScript;
+    private Script flyoverErpRedeemScript;
+    private Script p2shErpRedeemScript;
+    private Script flyoverP2shErpRedeemScript;
+    private List<Script> malformedScripts;
+    private List<Script> flyoverRedeemScripts;
+
+    @Before
+    public void setUp() {
+        final long CSV_VALUE = 52_560L;
+        standardRedeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+        flyoverStandardRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), standardRedeemScript);
+
+        erpRedeemScript = RedeemScriptUtils.createErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), erpRedeemScript);
+
+        p2shErpRedeemScript = RedeemScriptUtils.createP2shErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverP2shErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), p2shErpRedeemScript);
+
+        flyoverRedeemScripts = Arrays.asList(flyoverStandardRedeemScript, flyoverErpRedeemScript, flyoverP2shErpRedeemScript);
+
+        Script malformedScriptZeroSize = new Script(new byte[0]);
+        Script malformedScriptTwoSize = new Script(new byte[2]);
+
+        malformedScripts = Arrays.asList(malformedScriptZeroSize, malformedScriptTwoSize);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsValidRedeemScript_shouldReturnFlyoverMultiSigType() {
+        for(Script flyoverRedeemScript : flyoverRedeemScripts) {
+            // Arrange
+            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());
+
+            // Act
+            MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+            // Assert
+            assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+        }
+    }
```

Comment:
```suggestion
    @Test
    public void getMultiSigType_forStandardMultiSigScriptWithFlyover_shouldReturnFlyoverMultiSigType() {
        assertIsFlyoverMultiSigType(flyoverStandardRedeemScript);
        assertIsStandardMultiSigType(internalFlyoverRedeemScript);
    }
    
    @Test
    public void getMultiSigType_forErpMultiSigScriptWithFlyover_shouldReturnFlyoverMultiSigType() {
        assertIsFlyoverMultiSigType(flyoverErpRedeemScript);
    }
        
    @Test
    public void getMultiSigType_forP2shErpMultiSigScriptWithFlyover_shouldReturnFlyoverMultiSigType() {
        assertIsFlyoverMultiSigType(flyoverP2shErpRedeemScript);
    }
    
    private void assertIsFlyoverMultiSigType(Script flyoverRedeemScript) {
             // Arrange
            FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverRedeemScript.getChunks());

            // Act
            MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();

            // Assert
            assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
    }
    
    private void assertIsStandardMultiSigType(RedeemScript internalFlyoverRedeemScript) {
            assertEquals(MultiSigType.STANDARD_MULTISIG, actualMultiSigType);
    }
    
    ....
```

Reply:
same for the other tests

Reply:
No agree. Indeed, I applied this change to simulate the manual "`ParameterizedTest`" since in `bitcoinj-thin` we are using JUnit 4, and we are not able to use it. I applied it to avoid repeating code to test different types of Redeem Script in a test method. You can take a look at the commit with the specific changes.

Reply:
i know. But is not clean.
And btw, if you do it the way i propose, you get to test things related to the internal redeem scripts too because you know the case you are actually testing.

Reply:
and this is exactly what @marcos-iov proposed [here](https://github.com/rsksmart/bitcoinj-thin/pull/92#discussion_r1747202293).
His `testgetMultiSigType` method is my `assertIsFlyoverMultiSigType`.

Reply:
Well, then let me work on these changes.

Reply:
Done!
- Remove the "For" to iterate different kinds of Scripts since this is hard to follow.
- Split the test as per the script type and methods.